.github/workflows/docker-image.yml

```diff
@@ -188,17 +189,39 @@ jobs:
         working-directory: ./e2e
         run: npx playwright install chromium --with-deps
       - name: Run Playwright tests
+        id: run-playwright-tests
         working-directory: ./e2e
         run: npx playwright test
 
       - name: Upload screenshots as a github-pages artifact
         uses: actions/upload-pages-artifact@v3
-        if: failure()
+        # Only if the actual tests failed, if something else earlier in the
+        # job failed then the tests wouldn't have been run and the report
+        # would not exist to upload
+        if: failure() && steps.run-playwright-tests.outcome == 'failure'
         with:
           name: playwright-report
           path: e2e/playwright-report/
           retention-days: 7
 
+      # Need a way to communicate to the report-pages job whether or not we
+      # uploaded a report; only way that seems to be possible is to use an
+      # output from a step (here) and output that from the overall job (above)
+      # and then check that in the report-pages job (below). Since the overall
+      # output gets extracted from a single step, and steps need to have unique
+      # ids, cannot have two different steps that run under the two possible
+      # conditions, need to have a single step that checks the conditions
+      - name: Output if a report was uploaded
+        id: report-playwright-tests
+        if: always()
+        shell: bash
+        run: |
+          if ${{ steps.run-playwright-tests.outcome == 'failure' }}; then
+            echo "was-uploaded=yes" >> $GITHUB_OUTPUT
+          else
+            echo "was-uploaded=no" >> $GITHUB_OUTPUT
+          fi
```

Comment:
I must be missing something, but can't this be replaced by fetching the outcome of the `run-playwright-tests` step in the job outputs, like this:

```yaml
outputs:
      pages_url: ${{ steps.deployment_pages.outputs.page_url }}
      report-uploaded: ${{ steps.run-playwright-tests.outcome }}
```

and then referenced from the `report-pages` like this:

```yaml
if: failure() && (needs.deploy-e2e.outputs.report-uploaded == 'failure') && ( github.event_name == 'push' || github.event_name == 'pull_request' )
```

In other words, just capture the outcome of the `run-playwright-tests` step on `deploy-e2e` job into the job outputs and access the outputs from another job without an intermediate step?